File: Beecrowd/6 - Paradigms/1029 - Fibonacci, How Many Calls/1029_Fibonacci_How_Many_Calls.py
n = int(input())

for _ in range(n):
    x = int(input())

    fib_values = [0] * (x + 1)
    call_counts = [0] * (x + 1)

    fib_values[0] = 0
    fib_values[1] = 1
    call_counts[0] = 1
    call_counts[1] = 1

    for j in range(2, x + 1):
        fib_values[j] = fib_values[j - 1] + fib_values[j - 2]
        call_counts[j] = call_counts[j - 1] + call_counts[j - 2] + 1

    if x == 0:
        print(f"fib({x}) = 1 calls = {fib_values[x]}")
    else:
        print(f"fib({x}) = {call_counts[x] - 1} calls = {fib_values[x]}")

File: Beecrowd/6 - Paradigms/1029 - Fibonacci, How Many Calls/1029_Fibonacci_How_Many_Calls.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n;
    cin >> n;

    for (int i = 0; i < n; i++) {
        int x;
        cin >> x;

        vector<int> fibValues(x + 1, 0), callCounts(x + 1, 0);

        fibValues[0] = 0, fibValues[1] = 1;
        callCounts[0] = 1, callCounts[1] = 1;

        for (int j = 2; j <= x; j++) {
            fibValues[j] = fibValues[j - 1] + fibValues[j - 2];
            callCounts[j] = callCounts[j - 1] + callCounts[j - 2] +
                            1; // +1 for the current call
        }

        if (x == 0) {
            cout << "fib(" << x << ") = 1 calls = " << fibValues[x] << endl;
        } else {
            cout << "fib(" << x << ") = " << callCounts[x] - 1
                 << " calls = " << fibValues[x] << endl;
        }
    }

    return 0;
}